libraries/chain/include/eosio/chain/webassembly/eos-vm-oc/memory.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <eosio/chain/wasm_eosio_constraints.hpp>
+#include <eosio/chain/webassembly/eos-vm-oc/eos-vm-oc.hpp>
+#include <eosio/chain/webassembly/eos-vm-oc/intrinsic_mapping.hpp>
+
+#include <stdint.h>
+#include <stddef.h>
+
+#ifdef __clang__
+   #define GS_PTR __attribute__((address_space(256)))
+#else
+   #define GS_PTR __seg_gs
+#endif
+
+namespace eosio { namespace chain { namespace eosvmoc {
+
+class memory {
+      static constexpr uint64_t wasm_memory_size                  = eosio::chain::wasm_constraints::maximum_linear_memory;
+      static constexpr uint64_t intrinsic_count                   = boost::hana::length(intrinsic_table);
+      //warning: changing the following 3 params will invalidate existing PIC
+      static constexpr uint64_t mutable_global_size               = 8u  * eosio::chain::wasm_constraints::maximum_mutable_globals/4u;
+      static constexpr uint64_t table_size                        = 16u * eosio::chain::wasm_constraints::maximum_table_elements;
+      static constexpr size_t   wcb_allowance                     = 512u;
+      static_assert(sizeof(control_block) <= wcb_allowance, "EOS-VM OC memory doesn't set aside enough memory for control block");
+
+      //round up the prologue to multiple of 4K page
+      static constexpr uint64_t memory_prologue_size = ((memory::wcb_allowance + mutable_global_size + table_size + intrinsic_count*UINT64_C(8))+UINT64_C(4095))/UINT64_C(4096)*UINT64_C(4096);
```

Comment:
unfortunately querying _SC_PAGESIZE won't allow me to fold this in to a constexpr. is there anything else similar in a #define you're aware of?

Reply:
Right, sadly not.  We could add a constexpr variable in eos-vm in the next release and in some initializer place an assert that the two values match.  It won't give us any compile time protection, but it would at least ensure that the page size is correct with the system.